src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Editor/TempFileTextBufferManager.cs

```diff
@@ -148,9 +148,13 @@ protected override Task DisposeCoreAsync(bool initialized)
 
         public async Task SaveAsync()
         {
+            // We must clear the project dirty flag first. If it's dirty in memory, the ProjectReloadManager will detect that
+            // the project is dirty and fail the reload, and discard the changes.
+            var saveTask = _projectXmlAccessor.ClearProjectDirtyFlagAsync();
             var toWrite = await ReadBufferXmlAsync().ConfigureAwait(false);
+            await saveTask.ConfigureAwait(false);
```

Comment:
No, these are entirely separate operations that shouldn't affect each other. That's why I separated them, no need to wait to read the buffer while we wait for the project dirty flag to be cleared.